### PR TITLE
Login page and Dashboard

### PR DESCRIPTION
**Pull request by Kanishk3813**
Recoded the login page and dashboard in react. The login page now stores the credentials in firebase and the dashboard role credentials are stored in mongodb atlas.

### CODE REVIEW

**Comment by deepak4566** [comment]
Thanks  , but @Kanishk3813 also do work on that google ,github signup , i think u can use firebase google authentiation to do that stuff

File: package-lock.json
Comment (deepak4566) [inline package-lock.json:1]: do we really need this ?? package-lock.json??

**Comment by deepak4566** [comment]
@DigiReaper //

File: src/components/LoginPage.js [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+          <div className="text-center mt-6">
+            <button
+              onClick={() => setIsSignUp(!isSignUp)}
+              className="text-blue-500 hover:text-blue-700 font-bold transition duration-300"
+            >
+              {isSignUp ? 'Already have an account? Sign In' : 'Don’t have an account? Sign Up'}
+            </button>
+          </div>
+          <div className="text-center mt-6 flex justify-center space-x-4">
+            <a href="http://localhost:3000/auth/google" className="text-gray-700 hover:text-red-600 transition duration-300">
```
Comment (deepak4566) [inline src/components/LoginPage.js]: i see that u are able to use that signup at your local host right?

File: src/components/Dashboard.js [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+    setExternalIdInput(e.target.value);
+  };
+
+  const handleSubmit = async (e) => {
+    e.preventDefault();
+    const data = { service_name: serviceName, role_arn: roleArn, external_id: externalIdInput };
+    
+    try {
+      const response = await fetch('http://localhost:5000/create_role', {
+        method: 'POST',
```
Comment (deepak4566) [inline src/components/Dashboard.js:40]: same here , may be we can try other approach to get role details @DigiReaper  any feedback is appreciated

File: src/components/Dashboard.js [hunk 3]
```diff
@@ -0,0 +1,254 @@
...
+      }
+    } catch (error) {
+      setMessage('Error creating role');
+      console.error('Error:', error);
+    }
+  };
+
+  const redirectToAWSRoleCreation = () => {
+    const awsAccountNumber = '767398108291';
+    setExternalId(uuidv4());
```
Comment (deepak4566) [inline src/components/Dashboard.js]: yah for now , just for demo u can use this account , after we crete aws organisations we can use that account.

**Comment by deepak4566** [comment]
@DigiReaper yah u can check the deployment of this page from here 
 https://aws-login-page.vercel.app/

File: src/components/LoginPage.js (hunk 1 above)
Comment (Adithya-00) [reply src/components/LoginPage.js]: yeah lets make it environment independent

File: src/components/Dashboard.js (hunk 2 above)
Comment (Adithya-00) [reply src/components/Dashboard.js:40]: yeah same, we can't hard code the localhost url

**Pull request merged**